draft-irtf-cfrg-fiat-shamir.md

```diff
@@ -71,18 +71,24 @@ Where:
 
 # The Codec interface
 
+A codec is a collection of:
+- functions that map prover messages into the hash function domain,
+- functions that map hash outputs into verifier messages (of the desired distribution).
+In addition, the "init" function initializes the hash state with a session ID and an instance label.
+For byte-oriented codecs, this is just the concatenation of the two prefixed by their lengths.
+
 A codec provides the following interface.
 
     class Codec:
-        def new(iv: bytes) -> Codec
-        def prover_message(self, prover_message)
-        def verifier_challenge(self) -> verifier_challenge
+        def init(session_id, instance_label) -> hash_state
+        def prover_message(self, hash_state, elements)
+        def verifier_challenge(self, hash_state) -> verifier_challenge
 
 Where:
 
-- `init(iv: bytes) -> DuplexSponge` denotes the initialization function. This function takes as input a 32-byte initialization vector `iv` and initializes the state of the codec.
-- `prover_message(self, prover_message) -> self` denotes the absorb operation of the codec. This function takes as input a prover message `prover_message` and mutates the codec's internal state.
-- `verifier_challenge(self) -> verifier_challenge` denotes the squeeze operation of the codec. This function takes no inputs and uses the codec's internal state to produce an unpredictable verifier challenge `verifier_challenge`.
+- `init(session_id, instance_label) -> hash_state` denotes the initialization function. This function takes as input a session ID and an instance label, and returns the initial hash state.
```

Comment:
What is the session ID? Would I find out by reading the draft?

Reply:
[Section 4 has a pretty concise definition](https://mmaker.github.io/draft-irtf-cfrg-sigma-protocols/draft-irtf-cfrg-fiat-shamir.html#section-4-2.2.1):

> A session_id: the session identifier, for user-provided contextual information about the context where the proof is made (e.g. a URL, or a timestamp).